src/python/scripts/run_gerp.py

```diff
@@ -28,45 +28,75 @@
 
-    msa_file = argschema.fields.InputFile(
-        required=True, description="MSA file (MFA format)"
-    )
-    tree_file = argschema.fields.InputFile(
-        required=True, description="Tree file (Newick format). Must include every species in the MSA."
-    )
-    depth_threshold = argschema.fields.Float(
-        required=False,
-        description="Constrained elements' depth threshold for shallow columns, in substitutions "
-                    "per site. By default, 0.5."
-    )
-    gerp_exe_dir = argschema.fields.InputDir(
-        required=False,
-        description="Path where 'gerpcol' and 'gerpelem' binaries can be found. By default, resort to $PATH."
-    )
+    with open(ce_file) as ce_file_handler:
+        constraint_elements = [ce.split() for ce in ce_file_handler if ce.rstrip() != ""]
 
+    # enrich json file with constraint elements info
+    constraint_elems = []
+    for ce in constraint_elements:
+        constraint_elem = {}
+        constraint_elem["start"] = int(ce[0])
+        constraint_elem["end"] = int(ce[1])
+        constraint_elem["length"] = int(ce[2])
+        constraint_elem["score"] = ce[3]
+        constraint_elem["p-val"] = ce[4]
+        constraint_elems.append(constraint_elem)
+    align_set["constraint_elems"] = constraint_elems
 
-if __name__ == "__main__":
-    mod = argschema.ArgSchemaParser(schema_type=InputSchema)
+    with open(json_file, 'w') as json_file_handler:
+        json.dump(align_set, json_file_handler)
+
+
+def main(param: argparse.Namespace) -> None:
+    ''' Main function of the run_gerp.py script
+
+    This function is running gerpcol that define a gerpscore for every column of the genomic alignment bloc
+    and gerpelem that identify constraint elements across the alignment bloc
+
+    :param param: argparse.Namespace storing all the script parameters
+    :return: None
+    '''
 
-    cmd = ["gerpcol", "-t", mod.args["tree_file"], "-f", mod.args["msa_file"]]
-    if "gerp_exe_dir" in mod.args:
-        cmd[0] = os.path.join(mod.args['gerp_exe_dir'], cmd[0])
+    cmd = ["gerpcol", "-t", param.tree_file, "-f", param.msa_file]
+    cmd[0] = os.path.join(param.gerp_exe_dir, cmd[0])
     subprocess.run(cmd, check=True)
 
     # By default, gerpcol's ouput filename has the MSA filename plus ".rates" suffix
-    cmd = ["gerpelem", "-f", f"{mod.args['msa_file']}.rates"]
-    if "gerp_exe_dir" in mod.args:
-        cmd[0] = os.path.join(mod.args['gerp_exe_dir'], cmd[0])
-    if "depth_threshold" in mod.args:
-        cmd += ["-d", str(mod.args["depth_threshold"])]
+    cmd = ["gerpelem", "-f", f"{param.msa_file}.rates"]
+    cmd[0] = os.path.join(param.gerp_exe_dir, cmd[0])
+    cmd += ["-d", str(param.depth_threshold)]
     subprocess.run(cmd, check=True)
+
+    add_ce_to_json(f"{param.msa_file}.rates.elems", param.msa_file.replace(".fa", ".json"))
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Process some integers.')
```

Comment:
I think we can come up with something a little more informative than "Process some integers." :slightly_smiling_face: